optimum/intel/neural_compressor/modeling_base.py

```diff
@@ -0,0 +1,474 @@
+                else:
+                    model_class._keys_to_ignore_on_load_missing.extend(missing_keys_to_ignore_on_load)
+
+                try:
+                    model = model_class.from_pretrained(model_id, **kwargs)
+                except AttributeError:
+                    init_contexts = [no_init_weights(_enable=True)]
+                    with ContextManagers(init_contexts):
+                        model = model_class(config, **kwargs)
+
+                model_class._keys_to_ignore_on_load_unexpected = keys_to_ignore_on_load_unexpected
+                model_class._keys_to_ignore_on_load_missing = keys_to_ignore_on_load_missing
+
+                if state_dict_path is None:
+                    revision = model_kwargs.get("revision", None)
+                    if os.path.isdir(model_id):
+                        state_dict_path = os.path.join(model_id, file_name)
+                    elif os.path.isfile(model_id):
+                        state_dict_path = model_id
+                    else:
+                        local_files_only = False
+                        if is_offline_mode():
+                            logger.info("Offline mode: forcing local_files_only=True")
+                            local_files_only = True
+                        cache_dir = model_kwargs.get("cache_dir", None)
+                        if cache_dir is None:
+                            cache_dir = TRANSFORMERS_CACHE
+                        if isinstance(cache_dir, Path):
+                            cache_dir = str(cache_dir)
+                        try:
+                            state_dict_path = hf_hub_download(
+                                repo_id=model_id,
+                                filename=file_name,
+                                revision=revision,
+                                cache_dir=cache_dir,
+                                local_files_only=local_files_only,
+                            )
+                        except EnvironmentError as err:
+                            logger.error(err)
+                            msg = (
+                                f"Can't load config for '{model_id}'. Make sure that:\n\n"
+                                f"-'{model_id}' is a correct model identifier listed on 'https://huggingface.co/models'\n\n"
+                                f"-or '{model_id}' is a correct path to a directory containing a {file_name} file\n\n"
+                            )
+
+                            if revision is not None:
+                                msg += (
+                                    f"- or '{revision}' is a valid git identifier (branch name, a tag name, or a commit id) that "
+                                    f"exists for this model name as listed on its model page on 'https://huggingface.co/models'\n\n"
+                                )
+
+                            raise EnvironmentError(msg)
+
```

Comment:
`state_dict_path` can be given by using  `config.model_name_or_path` so we don't need all this in my opinion (+ should deprecated the `state_dict_path` argument)